prov/psm2/src/psmx2_tagged.c

```diff
@@ -1114,7 +1114,9 @@ static ssize_t psmx2_tagged_sendmsg(struct fid_ep *ep,
 	if (!msg || (msg->iov_count && !msg->msg_iov))
 		return -FI_EINVAL;
 
-	if (msg->iov_count > 1) {
+	if (msg->iov_count > PSMX2_IOV_MAX_COUNT) {
```

Comment:
Why not just make these asserts?

Reply:
Good question. It depends on if we want to catch the error in non-debug build. This specific patch is consistent with existing code. There are many other places in the provider that face the same question. They can be looked into together and handled in a separate PR.